src/_pytest/nodes.py

```diff
@@ -254,13 +261,13 @@ def getparent(self, cls):
     def _prunetraceback(self, excinfo):
         pass
 
-    def _repr_failure_py(self, excinfo, style=None):
-        # Type ignored: see comment where fail.Exception is defined.
-        if excinfo.errisinstance(fail.Exception):  # type: ignore
+    def _repr_failure_py(
+        self, excinfo: ExceptionInfo[Union[Failed, "FixtureLookupError"]], style=None
+    ) -> Union[str, ReprExceptionInfo, ExceptionChainRepr, "FixtureLookupErrorRepr"]:
```

Comment:
Same comment here re. returning unions -- if there's a way to simplify, would be better. Haven't checked this one, though.